File: article_demos/app/src/main/java/com/article/demos/MainActivity.java
package com.article.demos;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;

import com.alibaba.android.arouter.launcher.ARouter;
import com.article.demos.android.ui.AndroidFragment;
import com.article.demos.common.constant.Constant;
import com.article.demos.databinding.ActivityMainBinding;
import com.article.demos.java.ui.JavaFragment;
import com.article.demos.kotlin.ui.KotlinFragment;
import com.article.demos.vue.ui.VueFragment;
import com.flyco.tablayout.listener.CustomTabEntity;

import java.util.ArrayList;

/**
 * @author a
 */
public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding mBinding;
    private ArrayList<Fragment> mList = new ArrayList<>();
    private ArrayList<CustomTabEntity> mTabList = new ArrayList<>();

    private String[] mTitles = {"Android", "Java", "Kotlin", "Vue"};
    private int[] mIconSelectIds = {
            R.drawable.ic_android_selected, R.drawable.ic_java_sesected,
            R.drawable.ic_kotlin_selected, R.drawable.ic_vue_selected};
    private int[] mIconUnselectIds = {
            R.drawable.ic_android_unselected, R.drawable.ic_java_unselected,
            R.drawable.ic_kotlin_unselected, R.drawable.ic_vue_unselected};

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        initData();
    }

    private void initData() {
        JavaFragment java = (JavaFragment) ARouter.getInstance().build(Constant.JAVA_PAHT).navigation();
        AndroidFragment android = (AndroidFragment) ARouter.getInstance().build(Constant.ANDROID_PAHT).navigation();
        KotlinFragment kotlin = (KotlinFragment) ARouter.getInstance().build(Constant.KOTLIN_PAHT).navigation();
        VueFragment vue = (VueFragment) ARouter.getInstance().build(Constant.VUE_PAHT).navigation();

        mList.add(android);
        mList.add(java);
        mList.add(kotlin);
        mList.add(vue);

        for (int i = 0; i < mTitles.length; i++) {
            mTabList.add(new MainTabEntity(mTitles[i], mIconSelectIds[i], mIconUnselectIds[i]));
        }
        mBinding.tabLayout.setTabData(mTabList, this, R.id.frame_layout, mList);
    }
}

File: article_demos/wananzhuoapp/src/main/java/git/lxy/com/wananzhuoapp/main/MainActivity.kt
package git.lxy.com.wananzhuoapp.main

import android.arch.lifecycle.ViewModelProviders
import android.support.v4.app.Fragment
import com.alibaba.android.arouter.facade.annotation.Route
import com.article.demos.common.base.BaseActivity
import com.article.demos.common.constant.Constant
import com.flyco.tablayout.listener.CustomTabEntity
import git.lxy.com.wananzhuoapp.R
import git.lxy.com.wananzhuoapp.main.vm.WanAndroidMainViewModel
import git.lxy.com.wananzhuoapp.ui.tree.fragment.KnowledgeFragment
import git.lxy.com.wananzhuoapp.ui.my.fragment.MyFragment
import git.lxy.com.wananzhuoapp.ui.nav.NavigationFragment
import git.lxy.com.wananzhuoapp.ui.home.HomeFragment
import git.lxy.com.wananzhuoapp.ui.wechat.fragment.WeChatFragment
import kotlinx.android.synthetic.main.activity_wananzhuo_main.*

@Route(path = Constant.WAN_ANDROID_MAIN)
class MainActivity : BaseActivity() {

    private val mList = ArrayList<Fragment>()
    private val mTabList = ArrayList<CustomTabEntity>()

    private lateinit var mTitles: Array<String>
    private val mIconSelectIds = intArrayOf(R.drawable.wanandroid_ic_home_selected_24dp, R.drawable.wanandroid_ic_knowledge_selected, R.drawable.wanandroid_ic_navigation_selected, R.drawable.wananzhuo_ic_wechat_selected, R.drawable.wanandroid_ic_my_selected)
    private val mIconUnselectIds = intArrayOf(R.drawable.wanandroid_ic_home_unselected_24dp, R.drawable.wanandroid_ic_knowledge_unselected, R.drawable.wanandroid_ic_navigation_unselected, R.drawable.wananzhuo_ic_wechat_unselect, R.drawable.wanandroid_ic_my_unselected)

    lateinit var mViewModel: WanAndroidMainViewModel

    override fun getLayoutId() = R.layout.activity_wananzhuo_main
    override fun init() {
        mViewModel = ViewModelProviders.of(this).get(WanAndroidMainViewModel::class.java)
        initData()
    }

    private fun initData() {
        mTitles = resources.getStringArray(R.array.wanandroid_home_tab)

        mList.add(HomeFragment())
        mList.add(KnowledgeFragment())
        mList.add(NavigationFragment())
        mList.add(WeChatFragment())
        mList.add(MyFragment())

        for (i in mTitles.indices) {
            mTabList.add(WanAndroidMainTabEntity(mTitles[i], mIconSelectIds[i], mIconUnselectIds[i]))
        }
        tab_layout.setTabData(mTabList, this, R.id.frame_layout, mList)

        mViewModel.setData()
    }
}
